cdap-ui/graphql/resolvers/programRecordResolvers.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const programsResolver = {
```

Comment:
Can't this be directly `programRecordResolvers`? Do we need to declare another variable below?

Reply:
Yes, it can. However, if you would like to add more resolvers later, you would just need to merge it to the existing variable below. I can change it for now, if that is better.

Reply:
Also, I think it is better to export the set of "programRecordResolvers" instead of one specific resolver. That way, we need to only append new resolvers instead of appending then in addition to modifying all files that import the original resolver.

Reply:
I am not sure if I understand. What other files need to be changed? If we export a named resolver and if it needs to be used it has to be imported right? If its a single object that is exported we still need to change all those files to use the resolver right?

> Also, I think it is better to export the set of "programRecordResolvers" instead of one specific resolver. That way, we need to only append new resolvers instead of appending then in addition to modifying all files that import the original resolver.

Reply:
I am exporting `programRecordResolvers` in line 39 of this file. I am importing that object in `cdap-ui/graphql/graphql.js`. If we want to add more program resolvers, we would add them in this file and merge them into the `programRecordResolvers`; no additional changes needed. If I were to export the single `programsResolver`, I would have to import that object in `cdap-ui/graphql/graphql.js`. If we want to add more program resolvers in this file, then we cannot only export one resolver, but an object with all resolvers. Hence, I would have to change what object is exported in this file and change the import in `cdap-ui/graphql/graphql.js`.